Assets/Scripts/Resources/ResourceHolder.cs

```diff
@@ -16,21 +19,23 @@ public class ResourceHolder : MonoBehaviour
 	void Awake()
 	{
 		_transform = GetComponent<Transform>();
-		_beaconRenderer = GetComponentInChildren<BeaconTag>().gameObject.GetComponent<Renderer>();
+		_beaconRenderer = _beacon.GetComponent<Renderer>();
 		_particleSystem = GetComponentInChildren<ParticleSystem>();
 
 		RaycastHit hitInfo;
 
-		Physics.Raycast( new Ray( _transform.position, Vector3.up), out hitInfo, _maxHeight );
+		Physics.Raycast( new Ray( _transform.position, Vector3.up ), out hitInfo, _maxHeight, _collisionLayer );
+
+		float normalizedHeight = _maxHeight;
 
 		if ( hitInfo.transform )
 		{
-			// The .5 is to correct for some scale thing
-			float normalizedHeight = ( hitInfo.transform.position.y - _transform.position.y ) / ( _maxHeight * 0.5f );
-			_transform.localScale = _transform.localScale.SetY( normalizedHeight );
-			_particleSystem.startLifetime *= normalizedHeight;
+			normalizedHeight = hitInfo.transform.position.y - _transform.position.y;
 		}
 
+		_beacon.localScale = _beacon.localScale.SetY( normalizedHeight );
+		_particleSystem.startLifetime = normalizedHeight / _particleSystem.startSpeed * 0.5f; // when the speed of a particle is 1
```

Comment:
Does this depend on the speed? I.e if we speed up the particles but shorten the lifetime accordingly, does this line have to change?

The only reason I ask is because the comment makes me think that.